### ISSUE TITLE
PDF quality is bad, it's unreadable and swiping experience is not good

### ISSUE THREAD

**Issue by l2dev** [jidogoon/PdfRendererView#1]
For some reason pdfs look quite blurry

**Comment by jidogoon** [comment]
Quality problem is fixed in 1.0.5. The new version has just uploaded.
I'll investigate swiping problem. thanks.

**Comment by l2dev** [comment]
Nice!

How should I proceed with adding a progress message because now there is a delay (white window) before pdf is fetched.

**Comment by jidogoon** [comment]
> Nice!
> 
> How should I proceed with adding a progress message because now there is a delay (white window) before pdf is fetched.

I've updated [README](https://github.com/jidogoon/PdfRendererView/blob/master/README.md). You can set statusListener to show preparing status.

pdfView.statusListener = object: PdfRendererView.StatusCallBack {
    override fun onDownloadStart() {
        loading.visibility = View.VISIBLE
    }
    override fun onDownloadSuccess() {
        // downloading progress is not supported yet. will be added soon.
        loading.visibility = View.GONE
    }
    override fun onError(error: Throwable) {
        loading.visibility = View.GONE
    }
}

**Comment by l2dev** [comment]
Nice! Thanks for ur support

**Comment by l2dev** [comment]
Also what does it mean 

/// downloading progress is not supported yet. will be added soon.

As I see after downloading is finished event is triggered. Do you mean that reporting status in terms of live percentage is not available?

**Comment by jidogoon** [comment]
> Also what does it mean
> 
> /// downloading progress is not supported yet. will be added soon.
> 
> As I see after downloading is finished event is triggered. Do you mean that reporting status in terms of live percentage is not available?

Yes. That's what I mean. I just released new version and updated [README](https://github.com/jidogoon/PdfRendererView/blob/master/README.md).

**Comment by l2dev** [comment]
Great!

I also noticed more problems:

I have a multiple pdf with many files. If I zoom into one of these pdf then I can not scroll down anymore on remaining PDF's.

Also in the general overview when I keep scrolling down through a file with few pages the whole scrolling animation seems lagging.

Maybe you should try opening a pdf with google docs viewer and compare it against yours, would be easier to find these problems ?

**Comment by jidogoon** [comment]
> Great!
> 
> I also noticed more problems:
> 
> I have a multiple pdf with many files. If I zoom into one of these pdf then I can not scroll down anymore on remaining PDF's.
> 
> Also in the general overview when I keep scrolling down through a file with few pages the whole scrolling animation seems lagging.
> 
> Maybe you should try opening a pdf with google docs viewer and compare it against yours, would be easier to find these problems ?

As you said, I have to improve behavior on swipe, scrolling and zoom. So I already made a feature using google docs viewer. It is now working only for Kitkat. As you suggested, I'll add option using google docs viewer even when It's not kitkat. I'll notify you when the option added.

**Comment by l2dev** [comment]
> > Great!
> > I also noticed more problems:
> > I have a multiple pdf with many files. If I zoom into one of these pdf then I can not scroll down anymore on remaining PDF's.
> > Also in the general overview when I keep scrolling down through a file with few pages the whole scrolling animation seems lagging.
> > Maybe you should try opening a pdf with google docs viewer and compare it against yours, would be easier to find these problems ?
> 
> As you said, I have to improve behavior on swipe, scrolling and zoom. So I already made a feature using google docs viewer. It is now working only for Kitkat. As you suggested, I'll add option using google docs viewer even when It's not kitkat. I'll notify you when the option added.

would be really nice, thanks!

**Comment by l2dev** [comment]
@jidogoon  when can you add this flag for loading via google docs viewer?

**Comment by l2dev** [comment]
@jidogoon 

There is a problem with google doc viewer (it's not fault of your implementation) where sometimes document is not shown and there is a blank screen. So I predict that your library will become more and more popular.

**Comment by jidogoon** [comment]
@eurbon I've had some problems in development env. so I couldn't do something for new version. I just released new version. v1.0.7.

**Comment by l2dev** [comment]
@jidogoon  getting error: duplicate value for resource 'attr/showDivider' with config ''.
Message{kind=ERROR, text=error: duplicate value for resource 'attr/showDivider' with config ''., sources=[/Users/eimaurbon/.gradle/caches/transforms-1/files-1.1/appcompat-v7-28.0.0.aar/64cba8d69b0a6af81138ccd3ee31c5b8/res/values/values.xml:1304:5-69], original message=, tool name=Optional.of(AAPT)}
